src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -203,6 +203,31 @@ public function findByIds(array $ids): array
             ->getQuery()->getResult(Query::HYDRATE_ARRAY);
     }
 
+    public function findProgrammesByAncestryAndType(array $ancestryDbIds, string $entityType, ?int $limit, int $offset) : array
+    {
+        $this->assertEntityType($entityType, ['Clip', 'Episode', 'Series']);
+
+        $qb = $this->createQueryBuilder('programme')
+                   ->addSelect(['masterBrand', 'image', 'mbImage'])
+                   ->leftJoin('programme.image', 'image')
+                   ->leftJoin('programme.masterBrand', 'masterBrand')
+                   ->leftJoin('masterBrand.image', 'mbImage')
+                   ->where("programme INSTANCE OF ProgrammesPagesService:$entityType")
+                   ->andWhere('programme.ancestry LIKE :ancestry')
+                   ->andWhere('programme.streamable = 1')
+                   ->setFirstResult($offset)
+                   ->setMaxResults($limit)
+                   ->setParameter('ancestry', $this->ancestryIdsToString($ancestryDbIds) . '%');
+
+        if (in_array($entityType, ['Episode', 'Clip'])) {
+               // for programmeItems(clips, episode) return results ordered by streamableFrom
+               $qb->leftJoin("BBC\\ProgrammesPagesService\\Data\\ProgrammesDb\\Entity\\$entityType", 'programmeItem', 'WITH', 'programme = programmeItem');
```

Comment:
Why in the name of Satan's right testicle are you leftJoining core_entity to itself? That makes no sense to me.